src/exchange_manager.h

```diff
@@ -95,6 +95,8 @@ class ExchangeManager {
       typename std::set<Bid<T>*>::iterator it4;
       for (it4 = bids.begin(); it4 != bids.end(); ++it4) {
         Bid<T>* b = *it4;
+        Request<T>* r = b->request();
+        double pref = exctx.trader_prefs[b->bidder()][r][b];
```

Comment:
the answer is maybe.  It definitely changed the output in the right direction but not in the way I expected. Which may be an indication of a problem in my code.  . . . I’ll keep investigating.

-Meghan


On Jul 9, 2015, at 9:48 AM, Robert Carlsen <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/exchange_manager.h<https://github.com/cyclus/cyclus/pull/1191#discussion_r34262958>:

> @@ -95,6 +95,8 @@ class ExchangeManager {
>        typename std::set<Bid<T>*>::iterator it4;
>        for (it4 = bids.begin(); it4 != bids.end(); ++it4) {
>          Bid<T>* b = *it4;
> +        Request<T>* r = b->request();
> +        double pref = exctx.trader_prefs[b->bidder()][r][b];


@mbmcgarry<https://github.com/mbmcgarry> can you confirm if changing b->bidder() to r->requester fixes the zero pref issue?

—
Reply to this email directly or view it on GitHub<https://github.com/cyclus/cyclus/pull/1191/files#r34262958>.

*************************************************************************
Meghan McGarry, PhD ~ http://mbmcgarry.github.io/
Postdoctoral Fellow, Consortium for Verification Technology
Computational Nuclear Engineering ~ http://cnerg.engr.wisc.edu
Engineering Physics ~ UW-Madison ~ 608-263-6974
*************************************************************************